abs/replica_client.go

```diff
@@ -53,41 +62,86 @@ func (c *ReplicaClient) Init(ctx context.Context) (err error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	if c.containerURL != nil {
+	if c.client != nil {
 		return nil
 	}
 
-	// Read account key from environment, if available.
-	accountKey := c.AccountKey
-	if accountKey == "" {
-		accountKey = os.Getenv("LITESTREAM_AZURE_ACCOUNT_KEY")
-	}
-
-	// Authenticate to ACS.
-	credential, err := azblob.NewSharedKeyCredential(c.AccountName, accountKey)
-	if err != nil {
-		return err
+	// Validate required configuration
+	if c.Bucket == "" {
+		return fmt.Errorf("abs: container name is required")
 	}
 
 	// Construct & parse endpoint unless already set.
 	endpoint := c.Endpoint
 	if endpoint == "" {
+		if c.AccountName == "" {
+			return fmt.Errorf("abs: account name is required when endpoint is not specified")
+		}
 		endpoint = fmt.Sprintf("https://%s.blob.core.windows.net", c.AccountName)
 	}
-	endpointURL, err := url.Parse(endpoint)
-	if err != nil {
-		return fmt.Errorf("cannot parse azure endpoint: %w", err)
-	}
 
-	// Build pipeline and reference to container.
-	pipeline := azblob.NewPipeline(credential, azblob.PipelineOptions{
-		Retry: azblob.RetryOptions{
-			TryTimeout: 24 * time.Hour,
+	// Configure client options with retry policy
+	clientOptions := &azblob.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Retry: policy.RetryOptions{
+				MaxRetries:    10,
+				RetryDelay:    time.Second,
+				MaxRetryDelay: 30 * time.Second,
+				TryTimeout:    15 * time.Minute, // Reasonable timeout for blob operations
+				StatusCodes: []int{
+					http.StatusRequestTimeout,
+					http.StatusTooManyRequests,
+					http.StatusInternalServerError,
+					http.StatusBadGateway,
+					http.StatusServiceUnavailable,
+					http.StatusGatewayTimeout,
+				},
+			},
+			Telemetry: policy.TelemetryOptions{
+				ApplicationID: "litestream",
+			},
 		},
-	})
-	containerURL := azblob.NewServiceURL(*endpointURL, pipeline).NewContainerURL(c.Bucket)
-	c.containerURL = &containerURL
+	}
+
+	// Check if we have explicit credentials or should use default credential chain
+	accountKey := c.AccountKey
+	if accountKey == "" {
+		slog.Debug("no account key provided, using environment variable")
```

Comment:
One more nit... If the user is using credentials such as Managed Identity or from the Azure CLI, they're going to see this debug log message even though they're not _technically_ using env vars

Reply:
good point.  Updating error for clarity.  Thanks!